yaptide/celery/utils/manage_tasks.py

```diff
@@ -1,13 +1,68 @@
 import logging
+import redis
+import json
+import base64
 
 from celery import chord, group
 from celery.result import AsyncResult
 
 from yaptide.celery.tasks import merge_results, run_single_simulation
 from yaptide.celery.simulation_worker import celery_app
+from yaptide.persistence.db_methods import fetch_task_by_sim_id_and_task_id, update_task_state, update_simulation_state
 from yaptide.utils.enums import EntityState
 
 
+def stop_tasks_in_worker(simulation):
+    """Stops tasks that are running or are reserved in celery"""
+    ids_in_worker = get_tasks_from_celery(simulation_id=simulation.id)
+    task_ids = [task['task_id'] for task in ids_in_worker]
+    celery_ids = [task['celery_id'] for task in ids_in_worker]
+    tasks_celery = [fetch_task_by_sim_id_and_task_id(simulation.id, task_id) for task_id in task_ids]
+
+    result: dict = cancel_job(merge_id=simulation.merge_id, celery_ids=celery_ids)
+
+    if "merge" in result:
+        update_simulation_state(simulation=simulation, update_dict=result["merge"])
+        for task_index, task in enumerate(tasks_celery):
+            update_task_state(task=task, update_dict=result["tasks"][task_index])
+
+        return result
+    return None
+
+
+def decode_ids(redis_task_items):
+    """Helper function, decodes celery_id and task_id"""
+    array_ids = []
+    for item in redis_task_items:
+        item_data = json.loads(item)
+        body_encoded = item_data["body"]
+        decoded_body = base64.b64decode(body_encoded)
+        body_data = json.loads(decoded_body)
+        kwargs = body_data[1]
+        array_ids.append({"task_id": int(kwargs.get("task_id")), "simulation_id": int(kwargs.get("simulation_id"))})
+    return array_ids
+
+
+def delete_tasks_from_redis(simulation_id):
```

Comment:
what if we decide to reconfigure redis and use another (other than redis) storage backend ?